### ISSUE TITLE
TLS/SSL connection issues with LiberaChat

### ISSUE THREAD

**Issue by jaybeers** [kvirc/KVIrc#2510]
OK, so first off, a couple of disclaimers:

1. It says "LiberaChat" in the title, but really I don't think it's an issue with LiberaChat per se but with the TLS protocol negotiation in general.
2. This may be an Ubuntu build/packaging issue rather than something with KVIrc itself.

So, with that said:

I'm getting the following error when trying to connect to LiberaChat:

```
[2021-06-08 16:25:05] Attempting secure connection to tin.libera.chat (LiberaChat) on port 6697
[2021-06-08 16:25:05] Looking up the server hostname (tin.libera.chat)...
[2021-06-08 16:25:05] Server hostname resolved to 163.123.192.192
[2021-06-08 16:25:05] Contacting IRC server tin.libera.chat (163.123.192.192) on port 6697
[2021-06-08 16:25:05] Low-level transport connection established [tin.libera.chat (163.123.192.192:6697)]
[2021-06-08 16:25:05] Starting Secure Socket Layer handshake
[2021-06-08 16:25:06] [SSL ERROR]: wrong version number
[2021-06-08 16:25:06] [SOCKET ERROR]: Secure Socket Layer error
[2021-06-08 16:25:06] Connection attempt failed [tin.libera.chat]
[2021-06-08 16:25:06] Will attempt to reconnect in 10 seconds [10 of 15]
```

Usually it prints everything _except_ the `SSL ERROR` line (that's why I'm showing attempt 10 of 15).  I thought the cause was probably that KVIrc was built against an old version of OpenSSL that doesn't support newer TLS versions which LiberaChat might require, since it's all new and fancy.  However, I can connect to Freenode OK and when I do, I see this (along with all the other stuff):

```
[2021-06-08 16:34:14] Low-level transport connection established [card.freenode.net (38.229.70.22:6697)]
[2021-06-08 16:34:14] Starting Secure Socket Layer handshake
[2021-06-08 16:34:14] [SSL]: Current transmission cipher
[2021-06-08 16:34:14] [SSL]:  Cipher: TLS_AES_256_GCM_SHA384
[2021-06-08 16:34:14] [SSL]:  Version: TLSv1.3
[2021-06-08 16:34:14] [SSL]:  Bits: 256 (256 used)
[2021-06-08 16:34:14] Secure connection established [card.freenode.net (38.229.70.22:6697)]
```

So, supporting a new enough version of TLS 1.x is not an issue.  According to https://libera.chat/guides/connect the correct/recommended way is to connect with straight TLS on port 6697 (as opposed to making an initial cleartext connection to 6667 and doing STARTTLS, for example) but I've tried a bunch of different combinations and nothing gets me connected.  Is this a bug or a PEBCAK?  Here's the output from "Help > About KVIrc > Executable Information" (as packaged for Kubuntu 20.04):

```
KVIrc 5.0.0 'Aria'

Runtime Info:
System name: Linux 5.4.0-71-generic
System version: #79-Ubuntu SMP Wed Mar 24 10:56:57 UTC 2021
Architecture: x86_64
Qt version: 5.12.8
Qt theme: breeze

Build Info:
Build date: 2020-03-24 14:05:19 UTC
Sources date: 20160102
Revision number: 
CPU name: x86_64
Build command: /usr/bin/cmake
Build flags: 
   MANDIR=share/man
   CMAKE_INSTALL_PREFIX=/usr
   LIB_SUFFIX=/x86_64-linux-gnu
   Threads=POSIX
Compiler name: /usr/bin/c++
Compiler flags: -g -O2 -fdebug-prefix-map=/build/kvirc-tBiDRc/kvirc-5.0.0+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2
Qt version: 5.12.5
Features: IRC, Debug, IPv6, Crypt, SSL, IfAddr, IPC, KDE, Transparency, Phonon, Webkit, DCCVoice, Perl, Python, Enchant, Qt5, KVS
OpenSSL version: OpenSSL 1.1.1f 31 Mar 2020
OpenSSL compiler flags: compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -Wa,--noexecstack -g -O2 -fdebug-prefix-map=/build/openssl-olnknv/openssl-1.1.1f=. -fstack-protector-strong -Wformat -Werror=format-security -DOPENSSL_TLS_SECURITY_LEVEL=2 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=2
OpenSSL built on: Wed Apr 28 00:37:28 2021 UTC
```

**Comment by jaybeers** [comment]
OK, so just to close the loop on this: turns out it wasn't a KVIrc or LiberaChat issue at all, but rather a local (site-local, not my local host) firewalling issue.  I'd kinda ruled that out since it was connecting and then dropping, and I assumed an access control situation would just result in not being able to connect in the first place, but it turns out my understanding of how the firewalling in question works was incomplete.

So, long story short...DISREGARD!
